core/sr-io/src/lib.rs

```diff
@@ -93,40 +94,48 @@ export_api! {
 		fn storage(key: &[u8]) -> Option<Vec<u8>>;
 
 		/// Get `key` from child storage and return a `Vec`, empty if there's a problem.
-		fn child_storage(storage_key: &[u8], key: &[u8]) -> Option<Vec<u8>>;
+		fn child_storage(subtrie: SubTrieReadRef, key: &[u8]) -> Option<Vec<u8>>;
 
 		/// Get `key` from storage, placing the value into `value_out` (as much of it as possible) and return
 		/// the number of bytes that the entry in storage had beyond the offset or None if the storage entry
 		/// doesn't exist at all. Note that if the buffer is smaller than the storage entry length, the returned
 		/// number of bytes is not equal to the number of bytes written to the `value_out`.
 		fn read_storage(key: &[u8], value_out: &mut [u8], value_offset: usize) -> Option<usize>;
 
+		/// get child trie at storage key location
+		fn child_trie(prefix: &[u8], storage_key: &[u8]) -> Option<SubTrie>;
```

Comment:
Still reviewing but how should I use it, can I just store it in contract ? It seems I would just store the SubTrie instead of StorageKey.

Reply:
I did remove previously new fn `set_child_trie` to be closest to existing api:
the trie get created on first value update.
For the change I mention on contract (merging contract and the trie in a single node), a set_child_trie need to be created again: I am not doing it in this PR to assert that the `set_child_trie` is optional.